lint/shellcheck.bzl

```diff
@@ -72,21 +72,24 @@ def _shellcheck_aspect_impl(target, ctx):
     files_to_lint = filter_srcs(ctx.rule)
 
     if ctx.attr._options[LintOptionsInfo].fix:
-        patch, report, exit_code, info = patch_and_report_files(_MNEMONIC, target, ctx)
+        outputs, info = patch_and_output_files(_MNEMONIC, target, ctx)
         discard_exit_code = ctx.actions.declare_file(_OUTFILE_FORMAT.format(label = target.label.name, mnemonic = _MNEMONIC, suffix = "patch_exit_code"))
         if len(files_to_lint) == 0:
-            dummy_successful_lint_action(ctx, patch, discard_exit_code)
+            dummy_successful_lint_action(ctx, outputs.patch, discard_exit_code)
         else:
-            shellcheck_action(ctx, ctx.executable._shellcheck, files_to_lint, ctx.file._config_file, patch, discard_exit_code, ["--format", "diff"])
+            shellcheck_action(ctx, ctx.executable._shellcheck, files_to_lint, ctx.file._config_file, outputs.patch, discard_exit_code, ["--format", "diff"])
     else:
-        report, exit_code, info = report_files(_MNEMONIC, target, ctx)
+        outputs, info = output_files(_MNEMONIC, target, ctx)
 
     if len(files_to_lint) == 0:
-        dummy_successful_lint_action(ctx, report, exit_code)
+        dummy_successful_lint_action(ctx, outputs.human.stdout, outputs.human.exit_code)
     else:
         # shellcheck does not have a --fix mode that applies fixes for some violations while reporting others.
         # So we must run a second action to populate the human-readable report.
-        shellcheck_action(ctx, ctx.executable._shellcheck, files_to_lint, ctx.file._config_file, report, exit_code)
+        shellcheck_action(ctx, ctx.executable._shellcheck, files_to_lint, ctx.file._config_file, outputs.human.stdout, outputs.human.exit_code)
+
+    shellcheck_action(ctx, ctx.executable._shellcheck, files_to_lint, ctx.file._config_file, outputs.machine.stdout, outputs.machine.exit_code)
```

Comment:
no different in the human and machine actions for this linter? need a different pattern here to just copy the human outputs to the shell outputs instead of creating a separate action?

also, if `len(files_to_lint) == 0` then do a `dummy_successful_lint_action` for the machine action as well?

Reply:
There will be a difference in the future - the next PR adds the `--color` flag for the human report. https://github.com/koalaman/shellcheck/commit/a3d4101d6c9225582e6113288d898688f0a74e7f

Reply:
The dummy action was from a PR, I never really liked the way it was done. You have successfully prodded me into refactoring it to be a lot simpler, PTAL